pipelines/deployer/deployer.yaml

```diff
@@ -639,10 +639,72 @@ jobs:
     config: *unpack_release
     params:
       CLUSTER_PUBLIC_KEY: ((ci-system-gpg-public))
+  - task: scale-autoscaler-down
+    image: task-toolbox
+    timeout: 10s
+    params:
+      CLUSTER_NAME: ((cluster-name))
+      ACCOUNT_ROLE_ARN: ((account-role-arn))
+    config:
+      platform: linux
+      run:
+        path: /bin/bash
+        args:
+        - -euo
+        - pipefail
+        - -c
+        - |
+          echo "assuming aws deployer role..."
+          AWS_CREDS="$(aws-assume-role $ACCOUNT_ROLE_ARN)"
+          echo "Looking for existing EKS cluster"
+          CLUSTER_COUNT=$(aws eks list-clusters | jq '.clusters[] | select (. == "$CLUSTER_NAME") | 1' | wc -l)
+          if [ "$CLUSTER_COUNT" == "1" ]; then
+              echo "found it, fetching kubeconfig from aws..."
+              aws eks update-kubeconfig \
+                --name "${CLUSTER_NAME}" \
+                --kubeconfig ./kubeconfig
+              echo "done, looking for deployments"
+              export KUBECONFIG=$(pwd)/kubeconfig
+              DEPLOYMENT_COUNT=$(kubectl --namespace gsp-system get deployment -o json | jq '.items[] | select (.metadata.name == "gsp-aws-cluster-autoscaler") | 1' | wc -l)
+              if [ "$DEPLOYMENT_COUNT" == "1" ]; then
+                  echo "found it, scaling down to 0 replicas"
+                  kubectl --namespace gsp-system scale --replicas=0 deployment/gsp-aws-cluster-autoscaler
+                  echo "done"
+              fi
+          fi
   - put: cluster-state
     params:
       env_name: ((account-name))
       terraform_source: platform/pipelines/deployer
+  - task: scale-autoscaler-up
+    image: task-toolbox
+    timeout: 10s
+    params:
+      CLUSTER_NAME: ((cluster-name))
+      ACCOUNT_ROLE_ARN: ((account-role-arn))
+    config:
+      platform: linux
+      run:
+        path: /bin/bash
+        args:
+        - -euo
+        - pipefail
+        - -c
+        - |
+          echo "assuming aws deployer role..."
+          AWS_CREDS="$(aws-assume-role $ACCOUNT_ROLE_ARN)"
+          echo "fetching kubeconfig from aws..."
+          aws eks update-kubeconfig \
+            --name "${CLUSTER_NAME}" \
+            --kubeconfig ./kubeconfig
+          echo "done, looking for deployments"
+          export KUBECONFIG=$(pwd)/kubeconfig
+          DEPLOYMENT_COUNT=$(kubectl --namespace gsp-system get deployment -o json | jq '.items[] | select (.metadata.name == "gsp-aws-cluster-autoscaler") | 1' | wc -l)
+          if [ "$DEPLOYMENT_COUNT" == "1" ]; then
+              echo "found it, scaling up to 2 replicas"
+              kubectl --namespace gsp-system scale --replicas=2 deployment/gsp-aws-cluster-autoscaler
```

Comment:
We only have 1 replica of the cluster-autoscaler right now.  We should scale back up to where we scaled down from.

This means we either:

* explicitly scale up to 1 here
* change charts/gsp-cluster/values.yaml to set `cluster-autoscaler.replicaCount` to 2
* just rely on applying the chart to restore the cluster-autoscaler and remove this step?

I don't have a strong opinion on which of these options is best.

Reply:
Agreed, let's let the apply step restore normality.